### ISSUE TITLE
Retrieving html string from callback

### ISSUE THREAD

**Issue by MikePreston17** [iarna/rtf-to-html#3]
@iarna  Hi, this may not be a real issue, but I've been trying to get the html result out of the callback.  I'm relatively new to JS and Node just having started last week. :) I can log my sample callback below, but I can't retrieve your html in the callback you send to the pipe function.  What am I doing wrong?  I appreciate any help, thanks.

```
function sampleCallback() {
    var x = (function () {
        return true;
    })();
    console.log(x)
}

```
```
function usageSample(){
    let htmlOuter
    fs.createReadStream('example.rtf').pipe(rtfToHTML((err, html) => {
        console.log(html)
        htmlOuter = html
    }))
    console.log(htmlOuter)
}

```
```
function convertSample() {
    // console.log('started conversion')
    let callback = function (err, html) {
        console.log('callback-inner():\n', html)
        // this.html = html //I can log this.
        // return html //can't return this!
    }

    fs.createReadStream(letterPath).pipe(rtfToHtml(callback))

    console.log('callback-outer()', callback.html)
    var x = callback();
    console.log('x = ', x)
}
```

**Comment by MikePreston17** [comment]
fixes #3:
```

var onConversionComplete = new Event('on-convert-complete')

window.addEventListener('on-convert-complete', (e) => {
    // console.log('event received!')
    fs.readFile(savePath, 'utf8', (error, data) => {
        if (error) throw error;
        console.log("File red: ", savePath)
        ipc.send('html-data', data)
    })

}, false)

function convertIarnaSample() {
    console.log('started conversion')
    fs.createReadStream(letterPath).pipe(rtfToHtml((err, html) => {
        ipc.send('html-data', html)
        SaveHtml(html);
        window.dispatchEvent(onConversionComplete, html)
    }))
}

```

**Comment by iarna** [comment]
Ok, just trying to understand what you were running into…

Your first example, works as written:
```js
function usageSample(){
    let htmlOuter
    fs.createReadStream('example.rtf').pipe(rtfToHTML((err, html) => {
        console.log(html)
        htmlOuter = html
    }))
    console.log(htmlOuter)
}
```
 
Your second example does not:
```
function convertSample() {
    // console.log('started conversion')
    let callback = function (err, html) {
        console.log('callback-inner():\n', html)
        // this.html = html //I can log this.
        // return html //can't return this!
    }

    fs.createReadStream(letterPath).pipe(rtfToHtml(callback))

    console.log('callback-outer()', callback.html)
    var x = callback();
    console.log('x = ', x)
}
```
There are a few things about it. Let's start with the code after the callback. It'll never do anything interessting because streams are asynchronous. That means that when ` console.log('callback-outer()', …)` runs, nothing has been read from disk and no RTF has been processed. That code is run immediately, then, after `convertSample()` returns, the stream is processed and the callback is called. Ordinarily you'd put the code there _in the callback_. That's why the examples in the docs only have a callback block. You wouldn't ordinarily do anything outside of that. The callback is where execution continues from after parsing completes.

Anyway, it looks like you've found your way to another solution, so I'm gonna close this. If you run into anything else or want to follow up, please open a new issue (I probably won't see it if you just comment on this one). Thanks!
